crates/core/src/workflows/application.rs

```diff
@@ -8,15 +8,24 @@ use holochain_core_types::{
     dna::Dna,
     error::{HcResult, HolochainError},
 };
+use log::error;
 use std::sync::Arc;
 
 pub async fn initialize(
     instance: &Instance,
-    dna: Option<Dna>,
+    maybe_dna: Option<Dna>,
     context: Arc<Context>,
 ) -> HcResult<Arc<Context>> {
```

Comment:
What happens if we have a Dna in the context/state, but then call initialize with Some(dna)? If that is invalid behavior, maybe that is an argument for splitting this into two different functions so we can't do something invalid.

Reply:
Please have a look again. I've left it as one function for now, but added comments and a panic in case `maybe_dna` is `Some` AND there is a DNA already in the nucleus state.